core/src/main/scala/org/apache/spark/deploy/SparkSubmitArguments.scala

```diff
@@ -190,6 +190,7 @@ private[deploy] class SparkSubmitArguments(args: Seq[String], env: Map[String, S
       .orNull
     numExecutors = Option(numExecutors)
       .getOrElse(sparkProperties.get("spark.executor.instances").orNull)
+    queue = Option(queue).orElse(sparkProperties.get("spark.yarn.queue")).getOrElse("default")
```

Comment:
instead of `.getOrElse("default")`, i think it should be `orNull` - we don't know the default queue name of the system since it could change?